src/CssInliner.php

```diff
@@ -1093,15 +1123,22 @@ private function copyUninlinableCssToStyleNode(string $cssImportRules)
      * @param string $css
      *
      * @return void
+     *
+     * @throws \UnexpectedValueException if <head> was not found.
      */
     protected function addStyleElementToDocument(string $css)
     {
-        $styleElement = $this->domDocument->createElement('style', $css);
-        $styleAttribute = $this->domDocument->createAttribute('type');
+        $styleElement = $this->getDomDocument()->createElement('style', $css);
+        $styleAttribute = $this->getDomDocument()->createAttribute('type');
         $styleAttribute->value = 'text/css';
         $styleElement->appendChild($styleAttribute);
 
         $headElement = $this->getHeadElement();
+
+        if ($headElement === null) {
+            throw new \UnexpectedValueException('Could not find <head>');
```

Comment:
For other exceptions we have used a unique UNIX timestamp (the time at which the programmer put in the code, unique for each instance in the source code) as an exception code.  This could help precisely identify the source if an issue is reported.

Reply:
Oh, looks like I forgot to add the exception code. I'll need to do this (or someone else).

Reply:
> Oh, looks like I forgot to add the exception code. I'll need to do this (or someone else).

I've checked and all current (3) instances within `src` in `master` have a suitable code.  The only exceptions (literally, not `Exception`s) are cases where an `exception` is rethrown.

I was just commenting on the new addition(s) in this PR branch.

Reply:
noted, will get around to it after #795 is merged.

Reply:
I've added #797 for this.